pandas/io/parsers.py

```diff
@@ -1844,6 +1844,28 @@ def _do_date_conversions(self, names, data):
 
         return names, data
 
+    def _check_data_length(self, columns: List[str], data: List[np.ndarray]):
+        """Checks if length of data is equal to length of column names. One set of
+        trailing commas is allowed.
+
+        Parameters
+        ----------
+        columns: list of column names
+        data: list of array-likes containing the data column-wise
+
+        """
+        if not self.index_col and len(columns) != len(data) and columns:
```

Comment:
why do we need to check that data is actually null? IOW when would this situation happen when len(columns) > len(data) ?

Reply:
len(columns) > len(data) is caught at another place I think.
We run in there when len(columns) < len(data). In case of one set of trailing commas we have len(columns) + 1 = len(data). To see if we really have trailing commas we have to check if array is empty. If array is not empty we do not have trailing commas but data which will be dropped.

Reply:
> have trailing commas but data which will be dropped.

ok ideally we should put these kinds of checks in the same place that is happening if possible.

Reply:
Bad wording, with caught I meant if we got more columns than len(data), these columns are inserted all nans.